src/main/java/org/elasticsearch/index/engine/Segment.java

```diff
@@ -157,6 +168,37 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalBoolean(compound);
         out.writeOptionalString(mergeId);
         out.writeLong(memoryInBytes);
+        
+        boolean verbose = ramTree != null;
+        out.writeBoolean(verbose);
+        if (verbose) {
+            writeRamTree(out, ramTree);
+        }
+    }
+
+    Accountable readRamTree(StreamInput in) throws IOException {
+        final String name = in.readString();
+        final long bytes = in.readVLong();
+        int numChildren = in.readVInt();
+        if (numChildren == 0) {
+            return Accountables.namedAccountable(name, bytes);
+        }
+        List<Accountable> children = new ArrayList(numChildren);
+        while (numChildren-- > 0) {
+            children.add(readRamTree(in));
+        }
+        return namedAccountable(name, children, bytes);
+    }
+    
+    void writeRamTree(StreamOutput out, Accountable tree) throws IOException {
+        out.writeString(tree.toString());
+        out.writeVLong(tree.ramBytesUsed());
+        Iterable<Accountable> children = tree.getChildResources();
+        out.writeVInt(Iterators.size(children.iterator()));
```

Comment:
FYI I opened https://issues.apache.org/jira/browse/LUCENE-6148 so that we don't have to use Iterators.size